### ISSUE TITLE
mmplot_locator error

### ISSUE THREAD

**Issue by maxemil** [MadsAlbertsen/mmgenome#29]
Hej, I just started to try mmgenome and after successful installation i am now stuck in the basic step of selecting clusters. When I try to reproduce the 'genome_extraction' example and type `mmplot_locator(p)` i get the following error message:
```r
> sel <- mmplot_locator(p)
Error in grid.Call.graphics(L_downviewport, name$name, strict) : 
  Viewport 'panel.3-4-3-4' was not found
In addition: Warning messages:
1: Transformation introduced infinite values in continuous x-axis 
...
```

then I looked in the source of the function and saw that the name of the viewport was hardcoded as `panel.3-4-3-4`, but for some reason on my system the viewport seems to default to `layout`.... 
after changing the viewport name (saving it in a new function `my_locator`) I was able to set some selection points, but then received the following message when trying to select with the right mouse:

```
> sel <- my_locator(p)
Error in if (pi$panel$x_scales[[1]]$trans$name == "log-10") d[, 1] <- 10^(d[,  : 
  argument is of length zero
In addition: Warning messages:
1: Transformation introduced infinite values in continuous x-axis 
...
```

any idea on how to fix the problem would be highly appreciated!

I am running OpenSuse Leap 42.2 and my R version is 3.3.2.
Thanks!

edit: my sessionInfo:
```
> sessionInfo()
R version 3.3.2 (2016-10-31)
Platform: x86_64-suse-linux-gnu (64-bit)
Running under: openSUSE Leap 42.2

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] mmgenome_0.6.3      dplyr_0.5.0         reshape2_1.4.2     
 [4] igraph_1.0.1        Biostrings_2.42.0   XVector_0.14.0     
 [7] IRanges_2.8.1       S4Vectors_0.12.0    BiocGenerics_0.20.0
[10] vegan_2.4-1         lattice_0.20-34     permute_0.9-4      
[13] knitr_1.15          ggplot2_2.2.0       gridExtra_2.2.1    
[16] sp_1.2-3           

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.8      magrittr_1.5     cluster_2.0.5    zlibbioc_1.20.0 
 [5] MASS_7.3-45      munsell_0.4.3    colorspace_1.3-0 R6_2.2.0        
 [9] stringr_1.1.0    plyr_1.8.4       tools_3.3.2      gtable_0.2.0    
[13] nlme_3.1-128     mgcv_1.8-15      DBI_0.5-1        digest_0.6.10   
[17] lazyeval_0.2.0   assertthat_0.1   tibble_1.2       Matrix_1.2-7.1  
[21] labeling_0.3     stringi_1.1.2    scales_0.4.1    
```

**Comment by Kirk3gaard** [comment]
Hi maxemil

It is an issue with the new ggplot version. We are planning to make a fix ASAP. But in the meantime I recommend that you install an older version of ggplot

packageurl <- "http://cran.r-project.org/src/contrib/Archive/ggplot2/ggplot2_1.0.0.tar.gz"
install.packages(packageurl, repos=NULL, type="source")

you might need to install the package "proto" before (at least I had to)

**Comment by maxemil** [comment]
Hi,
Nice, that works like a charm, thanks for the help!

**Comment by aliexclusive** [comment]
Hi Kirk.

Thanks for the solution. I installed the package "proto" and then I installed older version ggplot as your recommended. But still it's giving the same error.

I'm using R Studio as platform.

Error in grid.Call.graphics(L_downviewport, name$name, strict) : 
  Viewport 'panel.3-4-3-4' was not found

Any idea why it's not working with me...

**Comment by Kirk3gaard** [comment]
Hi Aliexclusive

Can you provide the sessioninfo to ensure that the right version of the package is loaded?

**Comment by aliexclusive** [comment]
Here it is...


R version 3.3.2 (2016-10-31)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: macOS Sierra 10.12

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods  
[9] base     

other attached packages:
 [1] mmgenome_0.6.3      dplyr_0.5.0         reshape2_1.4.2      igraph_1.0.1       
 [5] vegan_2.4-1         lattice_0.20-34     permute_0.9-4       knitr_1.15         
 [9] ggplot2_1.0.0       gridExtra_2.2.1     sp_1.2-3            proto_1.0.0        
[13] Biostrings_2.42.0   XVector_0.14.0      IRanges_2.8.1       S4Vectors_0.12.0   
[17] BiocGenerics_0.20.0

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.8      cluster_2.0.5    magrittr_1.5     zlibbioc_1.20.0  MASS_7.3-45     
 [6] munsell_0.4.3    colorspace_1.3-1 R6_2.2.0         stringr_1.1.0    plyr_1.8.4      
[11] tools_3.3.2      grid_3.3.2       nlme_3.1-128     gtable_0.2.0     mgcv_1.8-16     
[16] DBI_0.5-1        assertthat_0.1   digest_0.6.10    tibble_1.2       Matrix_1.2-7.1  
[21] stringi_1.1.2    scales_0.4.1

**Comment by aliexclusive** [comment]
Hi Kirk3gaard.

It's now working. I tried to reload everything and now it's working but still an issue. It don't select the point where I click rather select somewhere else. Any idea why it is like this?

**Comment by Kirk3gaard** [comment]
Hi Aliexclusive

Great to hear that it helped.

No I have no idea why that would happen. Can you describe how to reproduce the problem? Otherwise it is impossible to track it down.

**Comment by aliexclusive** [comment]
![rplot](https://cloud.githubusercontent.com/assets/23704914/20608469/60f6343c-b293-11e6-917a-b51213dc0e60.png)

In this plot, I tried to extract genome related to plantomycetes on top left corner. But It draw area of interest (AOI) far away from the desired position.

**Comment by Kirk3gaard** [comment]
Hi Aliexclusive

That looks really weird. Can you post the code you used to generate this? We are aware of some issue that can cause the two layers to be out of sync (we are working on a fix or a workaround).

**Comment by aliexclusive** [comment]
Hi Kirk.

I'm using the following code. Basically using mmgenome guidelines.

p <- mmplot(data = d, x = "S1_2", y = "BS_granule", log.x = T, log.y = T, color = "essential", minlength = 3000)
p
sel <- mmplot_locator(p)

A good news is, it's working well when I try with R (Not R studio).

**Comment by Kirk3gaard** [comment]
Hi Ali

Thanks for the update. Great to hear that you found a solution.

**Comment by houjialin** [comment]
Hi aliexclusive:
I have the same error with you, and I installed the older version ggplot2 as the Kirk's suggestion and restarted, but the error also is 
Error in grid.Call.graphics(L_downviewport, name$name, strict) : 
  Viewport 'panel.3-4-3-4' was not found

could you give some suggestion else?

**Comment by aliexclusive** [comment]
Hi Houjialin. Are you using R or R Studio as platform? Can you please share sessioninfo...

**Comment by MadsAlbertsen** [comment]
## A small update

The problem is due to an update of the ggplot package and how they internally handle their data-structure, which ment that `mmplot_locater` broke. 

The "points are out of sync" problem seems to be related to when the axis are log-transformed. 

We have written a completely new function that should hopefully be live tomorrow, after we have had time to test it more extensively.

**Comment by houjialin** [comment]
Hi MadsAlbertsen:
I appreciate for your replying,  I can solve this problem by plotting the variance p before mmplot_locater function now.

Expecting for the new function
Thanks a lot
2016-11-28 


侯佳林



发件人：Mads Albertsen <notifications@github.com>
发送时间：2016-11-28 15:04
主题：Re: [MadsAlbertsen/mmgenome] mmplot_locator error (#29)
收件人："MadsAlbertsen/mmgenome"<mmgenome@noreply.github.com>
抄送："houjialin"<houjialin6@163.com>,"Comment"<comment@noreply.github.com>

A small update
The problem is due to an update of the ggplot package and how they internally handle their data-structure, which ment that mmplot_locater broke.
The "points are out of sync" problem seems to be related to when the axis are log-transformed.
We have written a completely new function that should hopefully be live tomorrow, after we have had time to test it more extensively.

**Comment by houjialin** [comment]
Hi aliexclusive:
      I can solve this problem by plotting the variance p before mmplot_locater function now.
Thanks for your help~

2016-11-28 


侯佳林



发件人：aliexclusive <notifications@github.com>
发送时间：2016-11-28 13:25
主题：Re: [MadsAlbertsen/mmgenome] mmplot_locator error (#29)
收件人："MadsAlbertsen/mmgenome"<mmgenome@noreply.github.com>
抄送："houjialin"<houjialin6@163.com>,"Comment"<comment@noreply.github.com>

Hi Houjialin. Are you using R or R Studio as platform? Can you please share sessioninfo...

**Comment by Gloriatorrescortes** [comment]
Hi, thanks a lot for all your help. I managed to fix my problem following your instructions. I was just wondering if finally the new function was release or not.
Thanks and regards!

**Comment by caonetto** [comment]
Hi, Thanks for the help, however I updated to the latest mmgenome and Im still having the issue of the "points are out of sync", any ways to fix this in rstudio??

Thanks again!

**Comment by MadsAlbertsen** [comment]
You need to manually tell the function which transformation you are using e.g.:

> mmplot_locator(p, trans = c("log10", "log10"))

We could not get a generic solution to work, hence the slightly more manual solution.

**Comment by caonetto** [comment]
Thanks Mads,
I tried with that already some time ago but it still points somewhere else in the plot. I just ended up writing the coordinates manually.

Thanks!


Cris


On 4 Apr. 2017, at 5:17 pm, Mads Albertsen <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #29<https://github.com/MadsAlbertsen/mmgenome/issues/29>.
